### PR TITLE
typing: classif

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/classification/accuracy.py [hunk 1]
```diff
@@ -227,10 +228,10 @@ def update(self, preds: Tensor, target: Tensor) -> None:
         """ returns the mode of the data (binary, multi label, multi class, multi-dim multi class) """
         mode = _mode(preds, target, self.threshold, self.top_k, self.num_classes, self.multiclass)
 
-        if self.mode is None:
+        if not self.mode:
```
Comment (justusschock) [inline torchmetrics/classification/accuracy.py:231]: can this also be some other value that evaluates to `False` (other then `None`)? If not I'd prefer the more explicit check with `is None`

File: torchmetrics/classification/accuracy.py [hunk 2]
```diff
@@ -227,10 +228,10 @@ def update(self, preds: Tensor, target: Tensor) -> None:
...
             self.mode = mode
         elif self.mode != mode:
-            raise ValueError("You can not use {} inputs with {} inputs.".format(mode, self.mode))
+            raise RuntimeError(f"You can not use {mode} inputs with {self.mode} inputs.")
```
Comment (justusschock) [inline torchmetrics/classification/accuracy.py:234]: A ValueError seems fine here...

File: torchmetrics/classification/accuracy.py [hunk 3]
```diff
@@ -240,6 +241,8 @@ def update(self, preds: Tensor, target: Tensor) -> None:
             self.correct += correct
             self.total += total
         else:
+            if not self.mode:
+                raise RuntimeError("You have to have determined mode.")
```
Comment (justusschock) [inline torchmetrics/classification/accuracy.py:245]: Is this even possible based on the conditions before?

File: torchmetrics/classification/accuracy.py [hunk 4]
```diff
@@ -269,6 +272,8 @@ def compute(self) -> Tensor:
         """
         Computes accuracy based on inputs passed in to ``update`` previously.
         """
+        if not self.mode:
```
Comment (justusschock) [inline torchmetrics/classification/accuracy.py:275]: Is this even possible based on the conditions before?

File: torchmetrics/classification/auroc.py [hunk 5]
```diff
@@ -138,7 +139,7 @@ def __init__(
                     '`max_fpr` argument requires `torch.bucketize` which is not available below PyTorch version 1.6'
                 )
 
-        self.mode = None
+        self.mode: DataType = None  # type: ignore
```
Comment (justusschock) [inline torchmetrics/classification/auroc.py:142]: Why can't we type this correctly?

File: torchmetrics/classification/accuracy.py (hunk 1 above)
Comment (Borda) [reply torchmetrics/classification/accuracy.py:231]: False is also not valid option

File: torchmetrics/classification/accuracy.py (hunk 3 above)
Comment (Borda) [reply torchmetrics/classification/accuracy.py:245]: yes if you call compute without any update

File: torchmetrics/classification/accuracy.py (hunk 4 above)
Comment (Borda) [reply torchmetrics/classification/accuracy.py:275]: yes if you call compute without any update

File: torchmetrics/classification/accuracy.py (hunk 3 above)
Comment (justusschock) [reply torchmetrics/classification/accuracy.py:245]: But we prevent that in the `__init__` of the super class I think where we assert that update was called, no?

Comment (Borda) [reply torchmetrics/classification/accuracy.py:245]: yes, by mypy dos not recognise this logic...

**Pull request merged**